packages/server/src/arpa_reporter/db/arpa-subrecipients.js

```diff
@@ -72,21 +60,54 @@ async function getRecipient(id, trns = knex) {
         .then((rows) => rows[0]);
 }
 
-async function findRecipient(uei = null, tin = null, trns = knex) {
+async function findRecipient(fieldType = null, value = null, trns = knex) {
     const tenantId = useTenantId();
     const query = baseQuery(trns).where('arpa_subrecipients.tenant_id', tenantId);
 
-    if (uei) {
-        query.where('arpa_subrecipients.uei', uei);
-    } else if (tin) {
-        query.where('arpa_subrecipients.tin', tin);
+    if (fieldType === 'uei') {
+        query.where('arpa_subrecipients.uei', value);
+    } else if (fieldType === 'tin') {
+        query.where('arpa_subrecipients.tin', value);
+    } else if (fieldType === 'name') {
+        query.where('arpa_subrecipients.name', value);
     } else {
         return null;
```

Comment:
It might be worth logging or printing something here or failing if a fieldType is passed in that doesn't match uei, tin or name.

We should also document the expected `fieldType` values in a docstring.

Reply:
good call- I updated docstrings as well as updated the method to throw an error instead of returning `null` as it can be confusing for the caller.